### PR TITLE
#7518 Fixed type mismatch between `EntityRepository#__construct()` and its documented constructor arguments

### PR DESCRIPTION
**Pull request by koftikes**

### CODE REVIEW

File: lib/Doctrine/ORM/EntityRepository.php [hunk 1]
```diff
@@ -58,8 +58,8 @@ class EntityRepository implements ObjectRepository, Selectable
     /**
      * Initializes a new <tt>EntityRepository</tt>.
      *
-     * @param EntityManager         $em    The EntityManager to use.
-     * @param Mapping\ClassMetadata $class The class descriptor.
+     * @param EntityManagerInterface    $em    The EntityManager to use.
```
Comment (Ocramius) [inline lib/Doctrine/ORM/EntityRepository.php]: Can be dropped completely

File: lib/Doctrine/ORM/EntityRepository.php [hunk 2]
```diff
@@ -58,8 +58,8 @@ class EntityRepository implements ObjectRepository, Selectable
...
+     * @param Mapping\ClassMetadata     $class The class descriptor.
```
Comment (Ocramius) [inline lib/Doctrine/ORM/EntityRepository.php]: Can be dropped completely (comment is not really telling anything useful)

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: lib/Doctrine/ORM/EntityRepository.php (hunk 2 above)
Comment (koftikes) [reply lib/Doctrine/ORM/EntityRepository.php]: You mean fix to that:
```
    /**
     * Initializes a new <tt>EntityRepository</tt>.
     *
     * @param EntityManagerInterface $em
     * @param Mapping\ClassMetadata  $class
     */
```

Comment (lcobucci) [reply lib/Doctrine/ORM/EntityRepository.php]: @koftikes nope, drop both `@param`. They aren't needed since types are declared properly.

Comment (koftikes) [reply lib/Doctrine/ORM/EntityRepository.php]: @lcobucci nope it's not correct params declarations need for normal code 
image

Comment (Ocramius) [reply lib/Doctrine/ORM/EntityRepository.php]: That's an IDE warning that we ignore: our coding standards enforce dropping any docblock entry that doesn't describe the actual purpose of the referenced symbol.

Comment (koftikes) [reply lib/Doctrine/ORM/EntityRepository.php]: really? you have all descriptions on each constructor. lol.

**Review by Ocramius — APPROVED** [review]
:+1:

**Comment by Ocramius** [comment]
Thanks @koftikes!

**Pull request merged**